### PR TITLE
VyOS: firewall_rules module added 

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: rohitthakur2590 <rohitthakur2590@outlook.com>

##### SUMMARY

Add new resource module vyos_firewall_rules.
Targets model - ansible-network/resource_module_models#70
[vyos_firewall_rules]

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

 vyos_firewall_rules
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

File: lib/ansible/modules/network/vyos/vyos_firewall_rules.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    - gathered
+    - rendered
+    - parsed
+    default: merged
+"""
+EXAMPLES = """
```
Comment (rohitthakur2590) [inline lib/ansible/modules/network/vyos/vyos_firewall_rules.py:370]: add some example .

File: lib/ansible/module_utils/network/vyos/config/firewall_rules/firewall_rules.py [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+        :param cmd: commands to be prepend.
+        :return: generated list of commands.
+        """
+        h_tcp = {}
+        commands = []
+        g_set = 'type'
+        if w[attr]:
+            key = 'flags'
+            rate = w[attr].get(key) or {}
+            if rate:
```
Comment (rohitthakur2590) [inline lib/ansible/module_utils/network/vyos/config/firewall_rules/firewall_rules.py]: need to change variable to flag

File: lib/ansible/module_utils/network/vyos/config/firewall_rules/firewall_rules.py [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+        return commands
+
+    def _add_rules(self, afi, name, w_rules, h_rules, opr=True):
+        """
+        This function forms the set/delete commands based on the 'opr' type
+        for rules attributes.
+        :param want: desired config.
+        :param have: target config.
+        :return: generated commands list.
+        """
```
Comment (rohitthakur2590) [inline lib/ansible/module_utils/network/vyos/config/firewall_rules/firewall_rules.py:272]: update description

File: test/units/modules/network/vyos/test_vyos_firewall_rules.py [hunk 4]
```diff
@@ -0,0 +1,1052 @@
...
+
+        self.mock_load_config = patch('ansible.module_utils.network.common.network.Config.load_config')
+        self.load_config = self.mock_load_config.start()
+
+        self.mock_get_resource_connection_config = patch('ansible.module_utils.network.common.cfg.base.get_resource_connection')
+        self.get_resource_connection_config = self.mock_get_resource_connection_config.start()
+
+        self.mock_get_resource_connection_facts = patch('ansible.module_utils.network.common.facts.facts.get_resource_connection')
+        self.get_resource_connection_facts = self.mock_get_resource_connection_facts.start()
+        self.mock_execute_show_command = patch('ansible.module_utils.network.vyos.facts.static_routes.static_routes.Static_routesFacts.get_device_data')
```
Comment (rohitthakur2590) [inline test/units/modules/network/vyos/test_vyos_firewall_rules.py]: not needed

**Review by NilashishC — APPROVED** [review]

**Review by justjais — APPROVED** [review]

**Pull request merged**